src/factory.py

```diff
@@ -0,0 +1,48 @@
+""" The factory module, with various types of factories """
+from random import choice
+import logging
+
+class Factory(object):
+    """ The Factory class, which holds the various factories as class methods """
+
+    @staticmethod
+    def enemyFactory():
+        """ The enemy factory method, which returns a random enemy
+            Precondition: None
+            Postcondition: Returns a randomly selected Enemy type"""
+
+        from src.enemy import ET, Enemy, Brownie, Drow, Goblin, Thug, Demon  # pylint: disable=unused-variable
+        enemyTypes = Enemy.__subclasses__()
+        pick = choice(enemyTypes)()
+        logging.debug("Creating new enemy of type %s", pick)
+        return pick
+
+    @staticmethod
+    def itemFactory():
+        """ The item factory method, for generating random Items
+            Precondition: None
+            Postcondition: Returns an instance of the randomly selected Item type """
+
+        from src.item import Item
+        #from consumable import Consumable # pylint: disable=unused-import
+        from src.weapon import Weapon # pylint: disable=unused-variable
+        itemTypes = Item.__subclasses__()
+        pick = itemTypes[choice(itemTypes)]()
+        logging.debug("Creating new item of type %s", pick)
```

Comment:
Is this just returning a string of the item type. Like it will return either "Weapon" or "Consumable"? Or do you want it to return an instance of Weapon/Consumable?

Reply:
It's returning an instance